crates/eips/src/eip2718.rs

```diff
@@ -45,6 +45,15 @@ impl From<alloy_rlp::Error> for Eip2718Error {
     }
 }
 
+impl From<Eip2718Error> for alloy_rlp::Error {
+    fn from(err: Eip2718Error) -> Self {
+        match err {
+            Eip2718Error::RlpError(err) => err,
+            Eip2718Error::UnexpectedType(_) => alloy_rlp::Error::Custom("Unexpected type flag"),
+        }
+    }
+}
+
 #[cfg(feature = "std")]
 impl std::error::Error for Eip2718Error {}
```

Comment:
unrelated: this lacks `fn source` which should return the wrapped rlperr

mind fixing this in a separate pr quickly?